### ISSUE TITLE
[Bug]: Initializing SBAS post-merge

### ISSUE THREAD

**Issue by SteffanDavies** [mobigroup/gmtsar#9]
If Python crashes after merge (such as during unwrapping due to lack of memory etc.), then a new SBAS object must be initialized in a new python terminal (with `force=False` to prevent wiping data). The problem is that it does not detect existing merged products in the workdir, and creates a dataframe with separate products (2, 3; 2, 3.. instead of 23; 23...)

To resume from previous state, it is necessary to run only part of `sbas.merge_parallel()` python code related to grouping, and ignore the rest related to processing, because it has already been done before.

**Comment by SteffanDavies** [comment]
I would suggest either:

1. Detect merge products during SBAS initialization using a flag or other way
2. Detect merge products during merge_parallel() using a flag or other way
4. Add a function like merge_parallel (get_mergedproducts()?) that only performes grouping of sbas.df

**Comment by mobigroup** [comment]
This is a design issue and I'm working on it for now. When we work on large rasters we can produce trans.dat tables and geocoding matrices for the separate subswaths but it crashes on the merged subswaths. The trans.dat  processing is included into the merging function and it produces the fault. I'm thinking about a separate geocoding processing function like to:
```
sbas.merge_parallel()
# call the function only when you need to convert the output grids to geographic coordinates
# or convert a landmark or other grid from geographic to radar coordinates
sbas.geocode_parallel()
```

And of course we need to have more robust geocoding processing. Using 32 bit unsigned integers we can stitch ~5 full scenes (2^32 ~= 4.3 B pixels) and it looks reasonable (and that's possible to allow 64 bit unsigned integers too.. if it's a real case to stitch more than 5 full scenes in 15m resolution). By my tests we can have the fast matrix geocoding even on the huge rasters using NetCDF trans.dat file instead of the unstructured binary file.

**Comment by mobigroup** [comment]
@SteffanDavies The issue is fixed in pygmtsar 2022.11.12.2 and Docker image 2022-11-12, see
https://pypi.org/project/pygmtsar/
https://hub.docker.com/r/mobigroup/pygmtsar
